redbot/cogs/audio/core/utilities/player.py

```diff
@@ -0,0 +1,671 @@
+            return True
+        except (IndexError, KeyError):
+            return False
+
+    async def _get_spotify_tracks(
+        self, ctx: commands.Context, query: Query, forced: bool = False
+    ) -> Union[discord.Message, List[lavalink.Track], lavalink.Track]:
+        if ctx.invoked_with in ["play", "genre"]:
+            enqueue_tracks = True
+        else:
+            enqueue_tracks = False
+        player = lavalink.get_player(ctx.guild.id)
+        api_data = await self._check_api_tokens()
+        if any([not api_data["spotify_client_id"], not api_data["spotify_client_secret"]]):
+            return await self.send_embed_msg(
+                ctx,
+                title=_("Invalid Environment"),
+                description=_(
+                    "The owner needs to set the Spotify client ID and Spotify client secret, "
+                    "before Spotify URLs or codes can be used. "
+                    "\nSee `{prefix}audioset spotifyapi` for instructions."
+                ).format(prefix=ctx.prefix),
+            )
+        elif not api_data["youtube_api"]:
+            return await self.send_embed_msg(
+                ctx,
+                title=_("Invalid Environment"),
+                description=_(
+                    "The owner needs to set the YouTube API key before Spotify URLs or "
+                    "codes can be used.\nSee `{prefix}audioset youtubeapi` for instructions."
+                ).format(prefix=ctx.prefix),
+            )
+        try:
+            if self.play_lock[ctx.message.guild.id]:
+                return await self.send_embed_msg(
+                    ctx,
+                    title=_("Unable To Get Tracks"),
+                    description=_("Wait until the playlist has finished loading."),
+                )
+        except KeyError:
+            pass
+
+        if query.single_track:
+            try:
+                res = await self.api_interface.spotify_query(
+                    ctx, "track", query.id, skip_youtube=True, notifier=None
+                )
+                if not res:
+                    title = _("Nothing found.")
+                    embed = discord.Embed(title=title)
+                    if query.is_local and query.suffix in _PARTIALLY_SUPPORTED_MUSIC_EXT:
+                        title = _("Track is not playable.")
+                        description = _(
+                            "**{suffix}** is not a fully supported "
+                            "format and some tracks may not play."
+                        ).format(suffix=query.suffix)
+                        embed = discord.Embed(title=title, description=description)
+                    return await self.send_embed_msg(ctx, embed=embed)
+            except SpotifyFetchError as error:
+                self.update_player_lock(ctx, False)
+                return await self.send_embed_msg(
+                    ctx, title=_(error.message).format(prefix=ctx.prefix)
```

Comment:
THIS IS PART 5/8 THAT NEED TO BE APPLIED TOGETHER WITH THIS.

```suggestion
                    ctx, title=error.message.format(prefix=ctx.prefix)
```